plugin/storage/es/factory.go

```diff
@@ -94,8 +94,7 @@ func NewFactoryWithConfig(
 		return nil, err
 	}
 
-	cfg.MaxDocCount = defaultMaxDocCount
-	cfg.Enabled = true
+	cfg.ApplyDefaults(&defaultConfig)
```

Comment:
Is it safe to call at this point? It is guaranteed not to override anything that was explicitly set?

Reply:
Yes, it looks okay because I checked by logging the config; it doesn't override anything; it just applies the default config.